src/Schema/Validator.php

```diff
@@ -4,173 +4,15 @@
 
 namespace OpenAPIValidation\Schema;
 
-use cebe\openapi\spec\Schema as CebeSchema;
-use OpenAPIValidation\Schema\Exception\ValidationKeywordFailed;
-use OpenAPIValidation\Schema\Keywords\AllOf;
-use OpenAPIValidation\Schema\Keywords\AnyOf;
-use OpenAPIValidation\Schema\Keywords\Enum;
-use OpenAPIValidation\Schema\Keywords\Items;
-use OpenAPIValidation\Schema\Keywords\Maximum;
-use OpenAPIValidation\Schema\Keywords\MaxItems;
-use OpenAPIValidation\Schema\Keywords\MaxLength;
-use OpenAPIValidation\Schema\Keywords\MaxProperties;
-use OpenAPIValidation\Schema\Keywords\Minimum;
-use OpenAPIValidation\Schema\Keywords\MinItems;
-use OpenAPIValidation\Schema\Keywords\MinLength;
-use OpenAPIValidation\Schema\Keywords\MinProperties;
-use OpenAPIValidation\Schema\Keywords\MultipleOf;
-use OpenAPIValidation\Schema\Keywords\Not;
-use OpenAPIValidation\Schema\Keywords\Nullable;
-use OpenAPIValidation\Schema\Keywords\OneOf;
-use OpenAPIValidation\Schema\Keywords\Pattern;
-use OpenAPIValidation\Schema\Keywords\Properties;
-use OpenAPIValidation\Schema\Keywords\Required;
-use OpenAPIValidation\Schema\Keywords\Type;
-use OpenAPIValidation\Schema\Keywords\UniqueItems;
-use function count;
+use cebe\openapi\spec\Schema;
+use OpenAPIValidation\Schema\Exception\SchemaMismatch;
 
-// This will load a whole schema and data to validate if one matches another
-class Validator
+interface Validator
```

Comment:
Can you explain the logic behind extracting this interface from the `SchemaValidator`? Looks like it is not used anywhere.

Reply:
Well, at least it can be used as a library public contract, allowing to mock it easily. I have other ideas but not sure about them

Reply:
It does not look like the implementation can be swapped with something else. I don't mind this interface, I wonder what are the other possible cases?

Reply:
well, it could be mocked and decorated using the interface. currently there is no points of extension for this class, so even we definalize it back in order to mock or decorate it we will get only FQCN and method signature. It's called interface, so I extracted it explicitly.

My other thoughts are about looking at other nested validator. Most of them really have the same signature, but a bit smudged. In general they take the a portion of `data` and the `schema` in some way and compare one with another. So signature (inlc. exceptions) stays the same.

If we incapsulate keyword knowledge into the keyword validator itself (like moving `isset` check inside it) and pass not keyword options but schema object - we will create incapsulated validators with the same signature, which has all knowled about data validation

Reply:
Sounds good to me